From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Brett T. Warden" <brett.t.warden@intel.com>
Date: Wed, 23 May 2018 18:07:20 -0700
Subject: [PATCH] Fixup makefiles

Replace overloaded use of CFLAGS in OCaml commands. Set and create
installation directory.
---
 src/Makefile                 |  6 +++---
 src/Makefile.OCaml           | 12 ++++++------
 src/fsmonitor/linux/Makefile |  2 +-
 3 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index f109b93..9a5d661 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -59,7 +59,7 @@ include Makefile.OCaml
 ######################################################################
 # Installation
 
-INSTALLDIR = $(HOME)/bin/
+INSTALLDIR = $(DESTDIR)/usr/bin/
 
 # This has two names because on OSX the file INSTALL shadows the target 'install'!
 install: doinstall
@@ -77,9 +77,9 @@ doinstall: buildexecutable
 	    echo "makefile variable NAME not bound"; \
 	    exit 1 \
 	  ; fi
-	-mv $(INSTALLDIR)/$(NAME)$(EXEC_EXT) /tmp/$(NAME)-$(shell echo $$$$)
+	mkdir -p $(INSTALLDIR)
 	cp $(NAME)$(EXEC_EXT) $(INSTALLDIR)
-	cp $(NAME)$(EXEC_EXT) $(INSTALLDIR)$(NAME)-$(MAJORVERSION)$(EXEC_EXT)
+	cp $(NAME)-fsmonitor$(EXEC_EXT) $(INSTALLDIR)
 
 ######################################################################
 # Demo
diff --git a/src/Makefile.OCaml b/src/Makefile.OCaml
index 6cfb069..60a0b78 100644
--- a/src/Makefile.OCaml
+++ b/src/Makefile.OCaml
@@ -259,7 +259,7 @@ endif
 # console when click-started
 #  ifeq ($(OSARCH), win32)
 #    COBJS+=winmain.c
-#    CFLAGS+=-cclib /subsystem:windows
+#    CAMLLINKFLAGS+=-cclib /subsystem:windows
 #  endif
 
 # Gtk GUI
@@ -311,7 +311,7 @@ DEP_INCLFLAGS+=-I fsmonitor -I fsmonitor/linux -I fsmonitor/solaris -I fsmonitor
 ### Static build setup
 
 ifeq ($(STATIC), true)
-  CFLAGS+=-cclib -static
+  CAMLLINKFLAGS+=-cclib -static
 endif
 
 ####################################################################
@@ -364,7 +364,7 @@ ifeq ($(DEBUGGING), false)
     ifneq ($(OSARCH), osx)
       # Strip the binary (does not work with MS compiler; might not work
       # under OSX)
-      CFLAGS+=-cclib -Wl,-s
+      CAMLLINKFLAGS+=-cclib -Wl,-s
     endif
   endif
 endif
@@ -436,13 +436,13 @@ fswatch.cmi : ubase/prefs.cmi
 
 $(NAME)$(EXEC_EXT): $(CAMLOBJS) $(COBJS)
 	@echo Linking $@
-	$(CAMLC) -verbose $(CAMLFLAGS) $(CAMLLDFLAGS) -o $@ $(CFLAGS) $(CAMLLIBS) $^ $(CLIBS)
+	$(CAMLC) -verbose $(CAMLFLAGS) $(CAMLLDFLAGS) -o $@ $(CAMLLINKFLAGS) $(CAMLLIBS) $^ $(CLIBS)
 
 # Unfortunately -output-obj does not put .o files into the output, only .cmx
 # files, so we have to use $(LD) to take care of COBJS.
 $(NAME)-blob.o: $(CAMLOBJS) $(COBJS)
 	@echo Linking $@
-	$(CAMLC) -dstartup -output-obj -verbose -cclib -keep_private_externs $(CAMLFLAGS) -o u-b.o $(CFLAGS) $(CAMLLIBS) $(CLIBS) $(CAMLOBJS)
+	$(CAMLC) -dstartup -output-obj -verbose -cclib -keep_private_externs $(CAMLFLAGS) -o u-b.o $(CAMLLINKFLAGS) $(CAMLLIBS) $(CLIBS) $(CAMLOBJS)
 	$(LD) -r -keep_private_externs -o $@ u-b.o $(COBJS)
 	$(RM) u-b.o
 
@@ -450,7 +450,7 @@ $(NAME)-blob.o: $(CAMLOBJS) $(COBJS)
 # Original:
 # $(NAME)-blob.o: $(CAMLOBJS) $(COBJS)
 # 	@echo Linking $@
-# 	$(CAMLC) -output-obj -verbose $(CAMLFLAGS) -o u-b.o $(CFLAGS) $(CAMLLIBS) $(CLIBS) $(CAMLOBJS)
+# 	$(CAMLC) -output-obj -verbose $(CAMLFLAGS) -o u-b.o $(CAMLLINKFLAGS) $(CAMLLIBS) $(CLIBS) $(CAMLOBJS)
 # 	$(LD) -r -o $@ u-b.o $(COBJS)
 # 	$(RM) u-b.o
 
diff --git a/src/fsmonitor/linux/Makefile b/src/fsmonitor/linux/Makefile
index 557994b..16e46f4 100644
--- a/src/fsmonitor/linux/Makefile
+++ b/src/fsmonitor/linux/Makefile
@@ -22,7 +22,7 @@ buildexecutable:: $(FSMONITOR)$(EXEC_EXT)
 
 $(FSMONITOR)$(EXEC_EXT): $(FSMCAMLOBJS) $(FSMCOBJS)
 	@echo Linking $@
-	$(CAMLC) -verbose $(CAMLFLAGS) $(CAMLLDFLAGS) -o $@ $(CFLAGS) $(FSMCAMLLIBS) $^ $(CLIBS)
+	$(CAMLC) -verbose $(CAMLFLAGS) $(CAMLLDFLAGS) -o $@ $(CAMLLINKFLAGS) $(FSMCAMLLIBS) $^ $(CLIBS)
 
 clean::
 	rm -f $(DIR)/*.cm[iox] $(DIR)/*.o $(DIR)/*~
